independent-projects/bootstrap/maven-resolver/src/test/java/io/quarkus/bootstrap/resolver/maven/options/BootstrapMavenOptionsTest.java

```diff
@@ -0,0 +1,46 @@
+package io.quarkus.bootstrap.resolver.maven.options;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class BootstrapMavenOptionsTest {
+    private static String toRestore;
+
+    @BeforeAll
+    static void recordPreviousCmdLineProperty() {
+        toRestore = System.getProperty(BootstrapMavenOptions.QUARKUS_INTERNAL_MAVEN_CMD_LINE_ARGS);
+    }
+
+    @AfterAll
+    static void restorePreviousCmdLineProperty() {
+        if (toRestore != null) {
+            System.setProperty(BootstrapMavenOptions.QUARKUS_INTERNAL_MAVEN_CMD_LINE_ARGS, toRestore);
+        } else {
+            System.setProperty(BootstrapMavenOptions.QUARKUS_INTERNAL_MAVEN_CMD_LINE_ARGS, "");
+        }
+    }
+
+    @Test
+    void updateMavenCmdLineAndExtractUserProperties() {
+        Map<Object, Object> props = new HashMap<>();
+        props.put("quarkus.args", "get -u foo");
+        props.put("foo", "bar boo");
+        props.put("empty", "");
+        props.put("BOOL", "true");
+        final List<String> got = BootstrapMavenOptions.updateMavenCmdLineAndExtractUserProperties("quarkus:dev " +
+                "-Dquarkus.args=get -u foo " +
```

Comment:
I don't think this is a correct way of setting these values. I think values with spaces should always be quoted.

Reply:
Hehe, yes but that's the string we get from `System.getenv("MAVEN_CMD_LINE_ARGS")`, that's why the test is checking that we return a properly quoted value even when getting garbage in…

Reply:
By the way, the test is in no way exhaustive, I'm just checking for most common cases here…